File: app/src/main/java/com/johntechinc/slotsmachinepoc/MyPageAdapter.java
package com.johntechinc.slotsmachinepoc;

import android.support.v4.view.PagerAdapter;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

/**
 * Created by ImL1s on 22/08/2017.
 * Description:
 */


public class MyPageAdapter extends PagerAdapter {
    private List<View> mViewList;

    public MyPageAdapter(List<View> mViewList) {
        this.mViewList = mViewList;
    }

    @Override
    public int getCount() {//必须实现
        return mViewList.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {//必须实现
        return view == object;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {//必须实现，实例化
        container.addView(mViewList.get(position));
        return mViewList.get(position);
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {//必须实现，销毁
        container.removeView(mViewList.get(position));
    }
}

File: app/src/main/java/com/johntechinc/slotsmachinepoc/MyPageAdapter.kt
package com.johntechinc.slotsmachinepoc

import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter

/**
 * Created by ImL1s on 22/08/2017.
 * Description: PagerAdapter for handling views in a ViewPager.
 */
class MyPageAdapter(private val mViewList: List<View>) : PagerAdapter() {

    override fun getCount(): Int {
        return mViewList.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view == `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        container.addView(mViewList[position])
        return mViewList[position]
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(mViewList[position])
    }
}
